### PR TITLE
[Fixed JENKINS-31357] TestRecorderRule to handle HeadlessException

### PR DESCRIPTION
**Pull request by tfennelly**
[JENKINS-31357](https://issues.jenkins-ci.org/browse/JENKINS-31357)

Happens on Mac.

### CODE REVIEW

**Comment by tfennelly** [comment]
@reviewbybees

File: src/main/java/org/jenkinsci/test/acceptance/recorder/TestRecorderRule.java [hunk 1]
```diff
@@ -62,29 +62,31 @@ protected void starting(Description description) {
...
+                    compressorName, outputDimension,
+                    bitDepth, quality, screenRate);
+
             this.screenRecorder = new JUnitScreenRecorder
                     (gc, gc.getBounds(), getFileFormat(mimeType),
                             outputFormatForScreenCapture, null, null, diagnostics);
             this.screenRecorder.start();
+        } catch (UnsupportedOperationException e) {
+            // E.g. java.awt.HeadlessException
+            logger.warn("Exception starting test recording {}", e);
```
Comment (oleg-nenashev) [inline src/main/java/org/jenkinsci/test/acceptance/recorder/TestRecorderRule.java]: Why don't you catch HeadlessException only?

**Comment by jglick** [comment]
[More readable diff](https://github.com/jenkinsci/acceptance-test-harness/pull/53/files?w=1)

File: src/main/java/org/jenkinsci/test/acceptance/recorder/TestRecorderRule.java (hunk 1 above)
Comment (jglick) [inline src/main/java/org/jenkinsci/test/acceptance/recorder/TestRecorderRule.java]: Or simply

```java
} catch (Exception e) {
    logger.warn("Exception starting test recording {}", e);
}
```

If you have some kind of visceral dislike of catching `Exception` (personally I have never understood what the problem is—we know that _any_ kind of exception ought to be caught here, and reported) you could use

```java
} catch (HeadlessException | IOException | AWTException e) {
    logger.warn("Exception starting test recording {}", e);
}
```

Comment (tfennelly) [inline src/main/java/org/jenkinsci/test/acceptance/recorder/TestRecorderRule.java]: I did originally and can put it back.

Comment (tfennelly) [inline src/main/java/org/jenkinsci/test/acceptance/recorder/TestRecorderRule.java]: > if you have some kind of visceral dislike of catching Exception

I assume that comment is for @oleg-nenashev?

@oleg-nenashev seems to prefer a more specific exception while @jglick prefers a more general. I'd probably prefer the more general, but don't think it's really worth debating so how about we just leave it as I have it?

**Comment by rsandell** [comment]
:bee:

**Pull request merged**